magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/AWSUtils.java

```diff
@@ -195,4 +197,27 @@ public static GetMetricStatisticsResponse getCloudwatchMetricStatistics( String
       return client.getMetricStatistics(request);
     }
   }
+
+  public static List<String> getS3AvailableSizeMetrics(String regionID, String bucketName, MagpieAWSClientCreator clientCreator) {
+
+    try (CloudWatchClient client = clientCreator.apply(CloudWatchClient.builder()).region(Region.of(regionID)).build()) {
+      List<String> availableMetrics = new ArrayList<>();
+      List<DimensionFilter> dimensions = Collections.singletonList(DimensionFilter.builder().name("BucketName").value(bucketName).build());
+
+      final var request = ListMetricsRequest.builder()
+        .namespace("AWS/S3")
+        .metricName("BucketSizeBytes")
+        .dimensions(dimensions)
+        .build();
+      ListMetricsResponse response = client.listMetrics(request);
+      for (Metric metric : response.metrics()) {
```

Comment:
Possible use of streams and collector here, but it doesn't read amazingly having tried:

      response.metrics()
        .forEach(metric -> metric.dimensions().stream().filter(dimension -> "StorageType".equals(dimension.name())).map(Dimension::value)
          .forEach(availableMetrics::add));

Reply:
I think with some line breaks and indentation your code reads better.

Thanks!